src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -571,22 +571,14 @@ internal static bool HasUnsuppressedErrors(DiagnosticBag diagnostics)
         {
             foreach (var diag in diagnostics.AsEnumerable())
             {
-                if (IsReportedError(diag))
+                if (diag.DefaultSeverity == DiagnosticSeverity.Error && !diag.IsSuppressed)
```

Comment:
>!diag.IsSuppressed [](start = 72, length = 18)

Do we need to check this condition? Can "errors by default" be suppressed?

Reply:
Yes, analyzer diagnostics which are errors by default can be suppressed in source by pragma/suppress message attributes. They cannot be programmatically suppressed as per the finalized design of DiagnosticSuppressor feature.

---
In reply to: [295479590](https://github.com/dotnet/roslyn/pull/36582#discussion_r295479590) [](ancestors = 295479590)

Reply:
They can be suppressed even with /nowarn. See [http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests/CommandLineTests.cs,9967](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.CommandLine.UnitTests/CommandLineTests.cs,9967)

---
In reply to: [295495458](https://github.com/dotnet/roslyn/pull/36582#discussion_r295495458) [](ancestors = 295495458,295479590)

Reply:
I am actually going to improve this check to `if (diag.DefaultSeverity == DiagnosticSeverity.Error && diag.IsNotConfigurable())`, which makes the semantics clear - return true on any error diagnostic that cannot be configured/suppressed. This includes all compiler diagnostics and analyzer errors that are guaranteed to break build.

---
In reply to: [295497929](https://github.com/dotnet/roslyn/pull/36582#discussion_r295497929) [](ancestors = 295497929,295495458,295479590)